checks/tools/io/cdo.py

```diff
@@ -21,60 +20,100 @@
+    valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu',
+                     'dom:mc', 'arolla:pn', 'tsa:pn',
+                     'eiger:mc', 'pilatus:mc',
+                     'manali:gpu-squashfs', 'balfrin:gpu-squashfs']
+    # TODO: update maintainers ?
+    maintainers = ['SO', 'CB']
+    tags = {'production', 'mch', 'external-resources'}
+
+    @run_after('init')
+    def setup_valid_pe(self):
+        modules = {
+            'arolla': ['cdo', 'netcdf-fortran'],
+            'tsa': ['cdo', 'netcdf-fortran'],
+            'eiger': ['CDO'],
+            'pilatus': ['CDO'],
+            'manali': ['cdo', 'netcdf-fortran'],
+            'balfrin': ['cdo', 'netcdf-fortran'],
+        }
+        if self.current_system.name in modules:
+            self.modules = modules[self.current_system.name]
+        else:
+            self.modules = ['CDO']
+
         if self.current_system.name in ['arolla', 'tsa']:
             self.exclusive_access = True
             self.valid_prog_environs = ['PrgEnv-gnu', 'PrgEnv-gnu-nompi']
-            self.modules = ['cdo', 'netcdf-fortran']
         elif self.current_system.name in ['eiger', 'pilatus']:
             self.valid_prog_environs = ['cpeGNU']
-            self.modules = ['CDO']
+        elif self.current_system.name in ['manali', 'balfrin']:
+            self.valid_prog_environs = ['PrgEnv-gnu']
+            self.squashfs_script = './exe.sh'
+            # until VCMSA-102 is resolved:
+            if self.current_system.name in ['balfrin']:
+                self.prerun_cmds += [
+                    r'echo "export LD_LIBRARY_PATH=\$LD_LIBRARY_PATH:'
+                    rf'\$SCRATCH/balfrin/lib64" >> {self.squashfs_script}'
+                ]
+            self.prerun_cmds += [
+                f'grep ^module rfm_job.sh |grep -v unuse >> '
+                f'{self.squashfs_script}',
+                f'echo "\\$*" >> {self.squashfs_script}',
+            ]
         else:
             self.valid_prog_environs = ['builtin']
-            self.modules = ['CDO']
-
-        self.maintainers = ['SO', 'CB']
-        self.tags = {'production', 'mch', 'external-resources'}
 
 
 # Check that the netCDF loaded by the CDO module supports the nc4 filetype
 # (nc4 support must be explicitly activated when the netCDF library is
 # compiled...).
 @rfm.simple_test
 class CDO_DependencyTest(CDOBaseTest):
-    def __init__(self):
-        super().__init__()
-        self.descr = ('verifies that the netCDF loaded by the CDO module '
-                      'supports the nc4 filetype')
-        self.sourcesdir = None
-        self.executable = 'nc-config'
-        self.executable_opts = ['--has-nc4']
-        self.sanity_patterns = sn.assert_found(r'^yes', self.stdout)
+    descr = ('verifies that the netCDF loaded by the CDO module '
+             'supports the nc4 filetype')
+    sourcesdir = None
+
+    @run_before('run')
+    def setup_per_partition_name(self):
+        if 'squashfs' in self.current_partition.name:
```

Comment:
Maybe it's better if we add a `squashfs` feature in the partition, rather than base it on the name of the name.

Reply:
https://github.com/eth-cscs/cscs-reframe-tests/pull/34/commits/9b382cddca9e2d926eb1316103ce249618f24bf0